src/execution/index/art/base_leaf.cpp

```diff
@@ -54,20 +56,20 @@ BaseLeaf<CAPACITY, TYPE> &BaseLeaf<CAPACITY, TYPE>::DeleteByteInternal(ART &art,
 
 void Node7Leaf::InsertByte(ART &art, Node &node, const uint8_t byte) {
 	// The node is full. Grow to Node15.
-	auto &n7 = Node::Ref<Node7Leaf>(art, node, NODE_7_LEAF);
+	NodeHandle<Node7Leaf> handle(art, node);
+	auto &n7 = handle.Get();
 	if (n7.count == CAPACITY) {
 		auto node7 = node;
 		Node15Leaf::GrowNode7Leaf(art, node, node7);
 		Node15Leaf::InsertByte(art, node, byte);
 		return;
 	}
```

Comment:
We likely need to change the logic here slightly, similar to what the `BaseNode` does (via `n7.count != CAPACITY`). If we look at `BaseNode::InsertChild`, the scope of the `NodeHandle` ends before growing the node. That way, we don't hold on to a node whose memory changes while growing. 

```cpp
{
		NodeHandle<Node4> handle(art, node);
		auto &n = handle.Get();

		if (n.count != CAPACITY) {
			InsertChildInternal(n, byte, child);
			return;
		}
	}
	// The node is full.
	// Grow to Node16.
	auto node4 = node;
	Node16::GrowNode4(art, node, node4);
	Node16::InsertChild(art, node, byte, child);
```